### ISSUE TITLE
[a11y] OptionsTile (wc): expand/collapse click event should occur on an interactive element

### ISSUE THREAD

**Issue by matthewgallo** [carbon-design-system/ibm-products#7544]
### Package

@carbon/ibm-products

### Browser

Chrome

### Operating System

MacOS

### Package version

Latest web component package

### React version

N/A

### Automated testing tool and ruleset

IBM Equal Access Checker

### Description

The expand/collapse behavior is not accessible via keyboard, the click event should happen on an interactive element.

### Reproduction/example

https://ibm-products-web-components.carbondesignsystem.com/?path=/story/experimental-optionstile--default

### Steps to reproduce

See [story](https://ibm-products-web-components.carbondesignsystem.com/?path=/story/experimental-optionstile--default) for options tile

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems
